### PR TITLE
Code readability and (basic) UX improvements

### PR DESCRIPTION
**Pull request by elia**
## Description

A few marginal changes targeting code readability and basic-level/common-sense UX improvements.

## Motivation and Context

Working on SolidusStripe these changes were nice-to-haves.

## How Has This Been Tested?

CI

## Screenshots

image

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.

### CODE REVIEW

File: templates/app/views/orders/_payment_info.html.erb [hunk 1]
```diff
@@ -12,20 +12,21 @@
...
-      <%= content_tag(:p, payment.display_amount) %>
-    <% else %>
-      <%= content_tag(:p, payment.payment_method.name) %>
-    <% end %>
+      <% else %>
+        <li><%= payment.payment_method.name %></li>
+      <% end %>
+      <li>
+        <strong><%= payment.display_amount %></strong>
+        (<%= payment.state %>)
```
Comment (kennyadsl) [inline templates/app/views/orders/_payment_info.html.erb:28]: Is the payment state rendered using i18n? If not, I think we should.

File: templates/app/views/checkouts/steps/payment_step/_payment_method_selector.html.erb [hunk 2]
```diff
@@ -6,7 +6,7 @@
           <%= radio_button_tag(
             "order[payments_attributes][][payment_method_id]",
             method.id,
-            method == @order.available_payment_methods.first,
+            method == methods.first,
```
[the remark this answers is not in the document]
Comment (elia) [reply templates/app/views/checkouts/steps/payment_step/_payment_method_selector.html.erb:9]: Yes, it's exactly the same content [as provided from outside](https://github.com/solidusio/solidus_starter_frontend/blob/d051343c72e7fbea42f79ffcac3309b031e768e9/templates/app/views/checkouts/steps/_payment_step.html.erb#L7)

File: templates/app/controllers/checkouts_controller.rb [hunk 3]
```diff
@@ -101,17 +102,20 @@ def massaged_params
...
-         (!params[:state] && !@order.has_checkout_step?(@order.state))
-        @order.state = 'cart'
-        redirect_to checkout_state_path(@order.checkout_steps.first)
-      end
-    end
+    return if skip_state_validation?
+    return if @order.has_checkout_step?(params[:state] || @order.state)
+
+    @order.state = 'cart'
+    redirect_to checkout_state_path(@order.checkout_steps.first)
```
[the remark this answers is not in the document]
Comment (elia) [reply templates/app/controllers/checkouts_controller.rb:109]: That would need also proper testing coverage and it seems a bit out of the scope of this PR, is it fine if I open an issue about doing that?

File: templates/app/controllers/checkouts_controller.rb [hunk 4]
```diff
@@ -101,17 +102,20 @@ def massaged_params
...
+  end
+
+  def ensure_valid_payment
+    return unless params[:state] == "confirm"
+    return unless @order.payment_required?
 
     # Fix for https://github.com/spree/spree/issues/4117
     # If confirmation of payment fails, redirect back to payment screen
```
[the remark this answers is not in the document]
Comment (elia) [reply templates/app/controllers/checkouts_controller.rb]: It was originally added here

https://github.com/spree/spree/commit/c26b7f6460bcf3f1e675a080aabbab183aa6f661#diff-35616ac20b57ee70dd8ef980eac73672c492d255b60014578e284fc6eadb0167R57-R63

I'll move it outside of the method, now that it's been extracted.

File: templates/app/controllers/checkouts_controller.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (kennyadsl) [reply templates/app/controllers/checkouts_controller.rb:109]: An issue would be great!

**Review by kennyadsl — APPROVED** [review]
👍

File: templates/app/controllers/checkouts_controller.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (elia) [reply templates/app/controllers/checkouts_controller.rb:109]: Done https://github.com/solidusio/solidus_starter_frontend/issues/322

File: templates/app/controllers/checkouts_controller.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (elia) [reply templates/app/controllers/checkouts_controller.rb]: Eventually moved it above `return unless params[:state] == "confirm"`

**Pull request merged**